### DISCUSSION TITLE
Feature request changeable `webcam fb_count`

### DISCUSSION THREAD

**Discussion by bclermont** [arendst/Tasmota#15199] [Q&A] [answered]
### PROBLEM DESCRIPTION

[Freenove FNK0060 is supported by Tasmota](https://github.com/blakadder/templates/commit/02cc3e064347070b69539a17e1907a4baabcb699#r68799953).

I tried the camera and it don't work. It just cause the stack IP to hang for minutes.

I figured what the problem is, and it's the `config.fb_count` is hardcoded for `2` [here](https://github.com/arendst/Tasmota/blob/e146c21/tasmota/xdrv_81_esp32_webcam.ino#L283).

I do confirm that that board support SPRAM but, as soon as I switch this to `1` it work.

That value is hardcoded to `2` for all SPRAM enabled board.

I'm not familiar with Tasmota codebase, is it possible to add a configuration to turn that value into an option/parameter?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Freenove FNK0060
- [x] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: VSCode & PlateformIO
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:

02:31:51.425 RSL: RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
02:31:51.646 RSL: RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
02:31:51.896 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
02:32:18.415 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_D42658","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
02:32:18.460 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Vbat power on reset","Uptime":"0T00:01:19","StartupUTC":"2022-03-21T01:30:59","Sleep":50,"CfgHolder":4617,"BootCount":37,"BCResetTime":"2022-03-16T06:34:09","SaveCount":47}}
02:32:18.484 RSL: STATUS2 = {"StatusFWR":{"Version":"11.0.0.4(webcam)","BuildDateTime":"2022-03-21T01:04:04","Core":"2_0_3","SDK":"v4.3-dev-7410-g40d126bfd1","CpuFrequency":240,"Hardware":"ESP32-D0WDQ6","CR":"389/699"}}
02:32:18.507 RSL: STATUS3 = {"StatusLOG":{"SerialLog":4,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"192.168.1.1","LogPort":514,"SSId":["Bandit",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A190000000000","00000080","00006000","00004000"]}}
02:32:18.533 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":1194,"Free":1856,"Heap":166,"StackLowMark":3,"PsrMax":4092,"PsrFree":4067,"ProgramFlashSize":4096,"FlashSize":4096,"FlashFrequency":80,"FlashMode":3,"Features":["00000809","0F9005CA","00040001","00000082","00000000","40000981","80000080","00200000","00000008"],"Drivers":"1,2,4,7,8,9,10,12,24,27,29,38,50,52,81","Sensors":"2,6,127"}}
02:32:18.567 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-D42658-1624","IPAddress":"192.168.1.107","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.0.1","DNSServer2":"0.0.0.0","Mac":"C8:C9:A3:D4:26:58","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
02:32:18.591 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_D42658","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
02:32:18.615 RSL: STATUS7 = {"StatusTIM":{"UTC":"2022-03-21T01:32:18","Local":"2022-03-21T02:32:18","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:50","Sunset":"19:03"}}
02:32:18.631 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-03-21T02:32:18","ESP32":{"Temperature":53.3},"TempUnit":"C"}}
02:32:18.649 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-03-21T02:32:18","Uptime":"0T00:01:19","UptimeSec":79,"Heap":166,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":31},"Wifi":{"AP":1,"SSId":"Bandit","BSSId":"E0:63:DA:03:4F:49","Channel":6,"Mode":"11n","RSSI":92,"Signal":-54,"LinkCount":1,"Downtime":"0T00:00:05"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:

N/A
```

### TO REPRODUCE
_Steps to reproduce the behavior:_

Use Tasmota latest version on FNK0060

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

webcam to initialize and streaming server to stream

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

N/A

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

N/A

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment, marked as answer]
Please, as requested, set the weblog to 4 and provide the bootlogs

**Reply by bclermont** [reply]
@barbudor you haven't read at the issue, I already fixed the code, just need to know why the value is hardcoded and how to provide the fix for other users

**Reply by barbudor** [reply]
I have perfectly read the issue and what you changed is NOT a solution
You disabled a key feature of the driver to use 2 framebuffers when PSRAM is supposed to be available

What we need to understand is why YOUR PSRAM is not working as expected
Please provide the boot logs with WebLog 4, or if your unit is not responsive do this through serial with serialog 4
Thanks

**Comment by bclermont** [comment]
> You disabled a key feature of the driver to use 2 framebuffers when PSRAM is supposed to be available
> What we need to understand is why YOUR PSRAM is not working as expected

Ah ok, @barbudor I'm sorry, I understand now

by the way, [Freenove's example is hardcoding fb_count to 1](https://github.com/Freenove/Freenove_ESP32_WROVER_Board/blob/main/C/Sketches/Sketch_05.1_CameraWebServer/Sketch_05.1_CameraWebServer.ino#L94)

Output of startup process with weblog 4:

```
00:00:00.002-206/48 HDW: ESP32-D0WDQ6 (PSRAM)
00:00:00.029-205/48 UFS: FlashFS mounted with 304 kB free
00:00:00.066 CFG: Loaded from File, Count 52
00:00:00.073 QPC: Count 1
00:00:00.075 CFG: CR 389/699, Busy 0
00:00:00.078 ROT: Mode 1
00:00:00.080 BRY: GC from 1479 to 1479 bytes, objects freed 0/1 (in 0 ms) - slots from 6/46 to 6/46
00:00:00.088 BRY: GC from 3616 to 2948 bytes, objects freed 7/20 (in 0 ms) - slots from 33/61 to 18/61
00:00:00.182 CFG: No '*.autoconf' file found
00:00:00.186 BRY: GC from 3929 to 3340 bytes, objects freed 3/29 (in 0 ms) - slots from 41/61 to 26/61
00:00:00.187 BRY: Berry initialized, RAM used=3340 bytes
00:00:00.208 BRY: No 'preinit.be'
00:00:00.210 SRC: Restart
00:00:00.213 Project tasmota - Tasmota Version 11.0.0.4(webcam)-2_0_3(2022-03-27T17:45:50)
00:00:00.292 BRY: No 'autoexec.be'
00:00:00.595 WIF: Checking connection...
00:00:00.595 WIF: Attempting connection...
00:00:00.038 WIF: Connecting to AP1 Bandit Channel 6 BSSId E0:63:DA:03:4F:49 in mode 11n as tasmota-D42658-1624...
00:00:03.445 WIF: Checking connection...
00:00:03.446 WIF: Connected
00:00:03.752 HTP: Web server active on tasmota-D42658-1624 with IP address 192.168.1.107
00:00:04.096 NTP: Sync time...
00:00:04.849 RTC: UTC 2022-03-27T17:46:36, DST 2022-03-27T02:00:00, STD 2022-10-30T03:00:00
18:46:36.000 RTC: Synced by NTP
18:46:37.360 RSL: INFO1 = {"Info1":{"Module":"Freenove-WROVER-Cam","Version":"11.0.0.4(webcam)","FallbackTopic":"cmnd/DVES_D42658_fb/","GroupTopic":"cmnd/tasmotas/"}}
18:46:37.373 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-D42658-1624","IPAddress":"192.168.1.107"}}
18:46:37.385 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":42}}
18:46:37.391 TFS: File 'autoexec.bat' not found
18:46:39.066 QPC: Reset
18:46:41.059 APP: Boot Count 42
18:46:42.012 CFG: Saved, Count 53, Bytes 4096
18:46:42.018 BRY: GC from 4815 to 3323 bytes, objects freed 17/31 (in 1 ms) - slots from 60/76 to 32/76
18:46:42.021 RSL: STATE = {"Time":"2022-03-27T18:46:42","Uptime":"0T00:00:11","UptimeSec":11,"Heap":170,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":31},"Wifi":{"AP":1,"SSId":"Bandit","BSSId":"E0:63:DA:03:4F:49","Channel":6,"Mode":"11n","RSSI":100,"Signal":-48,"LinkCount":1,"Downtime":"0T00:00:04"}}
18:46:42.058 RSL: SENSOR = {"Time":"2022-03-27T18:46:42","ESP32":{"Temperature":45.0},"TempUnit":"C"}
18:46:47.116 RSL: config = {"ip":"192.168.1.107","dn":"Tasmota","fn":["Tasmota",null,null,null,null,null,null,null],"hn":"tasmota-D42658-1624","mac":"C8C9A3D42658","md":"Freenove-WROVER-Cam","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"11.0.0.4","t":"tasmota_D42658","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[0,0,0,0],"ver":1}
18:46:47.182 RSL: sensors = {"sn":{"Time":"2022-03-27T18:46:47","ESP32":{"Temperature":53.3},"TempUnit":"C"},"ver":1}
18:46:56.242 CMD: wc init
18:46:56.243 SRC: Serial
18:46:56.244 CMD: Grp 0, Cmd 'WC', Idx 1, Len 4, Pld -99, Data 'init'
18:46:56.248 RSL: RESULT = {"WC":{"Stream":1,"Resolution":6,"Mirror":0,"Flip":0,"Saturation":-2,"Brightness":1,"Contrast":0,"Rtsp":0}}
18:46:56.793 WIF: Checking connection...
18:47:09.344 HTP: Main Menu
18:47:09.409 CAM: Stream init
18:47:09.412 CAM: User template
18:47:09.413 CAM: XCLK on GPIO 21 using ledc channel 0
18:47:09.414 CAM: PSRAM found
18:47:09.753 CAM: Initialized
```

after a little bit the ip stack is jam and ping fail for many minutes

**Reply by s-hadinger** [reply]
Do you have serial attached to the device? We're missing information about how the device is failing: crash with stackdump? Memory corruption? Can you please paste here logs when the device is failing.

**Reply by bclermont** [reply]
@s-hadinger this is the output when I'm using vscode to build and deploy the source code using platform.io

This is the output directly from serial port.

there is no stacktrace, ping stop replying for 2-3 minutes and the device come back online later

**Reply by s-hadinger** [reply]
If I understand well, the line `18:47:09.753 CAM: Initialized` is the last one showing on serial? I'm also curious what is output on serial when the device comes back online.

**Comment by bclermont** [comment]
Well, that is a bit odd.

I reverted back to original code to perform a test with `weblog 4` on.

And later, I started to customize the firmware to remove some features and turn on some others... and now it just work.
I can even use the maximum resolution size. Which didn't worked previously, I was capped at 640x480.



`platformio_tasmota_cenv.ini`:

```
[env:tasmota32-verycustom]
framework                   = ${common.framework}
platform                    = ${core32.platform}
platform_packages           = ${core32.platform_packages}
board_build.filesystem      = ${common.board_build.filesystem}
custom_unpack_dir           = ${common.custom_unpack_dir}
board                       = esp32_4M
upload_port                 = /dev/ttyUSB0
upload_speed                = 921600
upload_resetmethod          = ${common.upload_resetmethod}
extra_scripts               = ${esp32_defaults.extra_scripts}
build_unflags               = ${core32.build_unflags}
build_flags                 = ${core32.build_flags}
lib_ldf_mode                = ${common.lib_ldf_mode}
lib_compat_mode             = ${common.lib_compat_mode}
lib_extra_dirs              = ${common.lib_extra_dirs}
                              lib/libesp32
lib_ignore                  =
                              Servo(esp8266)
                              ESP8266AVRISP
                              ESP8266LLMNR
                              ESP8266NetBIOS
                              ESP8266SSDP
                              SP8266WiFiMesh
                              Ethernet(esp8266)
                              GDBStub
                              TFT_Touch_Shield_V2
                              ESP8266HTTPUpdateServer
                              ESP8266WiFiMesh
                              EspSoftwareSerial
                              SPISlave
                              Hash
                              ArduinoOTA
```

and:  `user_config_override.h`

```
#ifndef _USER_CONFIG_OVERRIDE_H_
#define _USER_CONFIG_OVERRIDE_H_

#define ENABLE_RTSPSERVER
#define USE_BMP
#define USE_MAX31855
#define USE_SGP30
#define USE_WEBCAM
#undef FIRMWARE_BLUETOOTH
#undef FIRMWARE_CLASSIC
#undef FIRMWARE_DISPLAYS
#undef FIRMWARE_IR
#undef FIRMWARE_KNX_NO_EMULATION
#undef FIRMWARE_LITE
#undef FIRMWARE_LVGL
#undef FIRMWARE_MINICUSTOM
#undef FIRMWARE_MINIMAL
#undef FIRMWARE_SENSORS
#undef FIRMWARE_TASMOTA32
#undef FIRMWARE_WEBCAM
#undef FIRMWARE_ZBBRIDGE
#undef MQTT_HOST_DISCOVERY
#undef ROTARY_V1
#undef USE_AC_ZERO_CROSS_DIMMER
#undef USE_ADE7953
#undef USE_ADS1115
#undef USE_ALEXA_AVS
#undef USE_ARILUX_RF
#undef USE_ARMTRONIX_DIMMERS
#undef USE_AS3935
#undef USE_AUTOCONF
#undef USE_BERRY
#undef USE_BH1750
#undef USE_BL09XX
#undef USE_BME68X
#undef USE_BUZZER
#undef USE_BUZZER
#undef USE_CCS811_V2
#undef USE_CSE7766
#undef USE_DDS2382
#undef USE_DDSU666
#undef USE_DEEPSLEEP
#undef USE_DEEPSLEEP
#undef USE_DEVICE_GROUPS
#undef USE_DHT
#undef USE_DHT12
#undef USE_DISCOVERY
#undef USE_DISPLAY
#undef USE_DISPLAY_ILI9341
#undef USE_DISPLAY_LCD
#undef USE_DISPLAY_MATRIX
#undef USE_DISPLAY_MODES1TO5
#undef USE_DISPLAY_SEVENSEG
#undef USE_DISPLAY_SSD1306
#undef USE_DOMOTICZ
#undef USE_DS1624
#undef USE_DS18x20
#undef USE_EMULATION_HUE
#undef USE_EMULATION_WEMO
#undef USE_ENERGY_DUMMY
#undef USE_ENHANCED_GUI_WIFI_SCAN
#undef USE_EXS_DIMMER
#undef USE_HIH6
#undef USE_HLW8012
#undef USE_HOME_ASSISTANT
#undef USE_HPMA
#undef USE_HRE
#undef USE_HRXL
#undef USE_HTU
#undef USE_HX711
#undef USE_IAQ
#undef USE_IBEACON
#undef USE_INA219
#undef USE_INFLUXDB
#undef USE_IR_RECEIVE
#undef USE_IR_REMOTE
#undef USE_KNX
#undef USE_KNX_WEB_MENU
#undef USE_LIGHT
#undef USE_LM75AD
#undef USE_LMT01
#undef USE_LVGL
#undef USE_LVGL_FREETYPE
#undef USE_MCP39F501
#undef USE_MGS
#undef USE_MHZ19
#undef USE_MI_ESP32
#undef USE_MIBLE
#undef USE_MP3_PLAYER
#undef USE_MQTT_AWS_IOT
#undef USE_MQTT_TLS
#undef USE_NOVA_SDS
#undef USE_PING
#undef USE_PMS5003
#undef USE_PN532_HSU
#undef USE_PS_16_DZ
#undef USE_PWM_DIMMER
#undef USE_PZEM_AC
#undef USE_PZEM_DC
#undef USE_PZEM004T
#undef USE_RC_SWITCH
#undef USE_RDM6300
#undef USE_RF_SENSOR
#undef USE_RULES
#undef USE_SCD30
#undef USE_SCD40
#undef USE_SCRIPT
#undef USE_SDM120
#undef USE_SDM230
#undef USE_SDM630
#undef USE_SDM72
#undef USE_SENDMAIL
#undef USE_SENSEAIR
#undef USE_SERIAL_BRIDGE
#undef USE_SERIAL_BRIDGE
#undef USE_SHELLY_DIMMER
#undef USE_SHT
#undef USE_SHT3X
#undef USE_SHUTTER
#undef USE_SHUTTER
#undef USE_SONOFF_D1
#undef USE_SONOFF_IFAN
#undef USE_SONOFF_RF
#undef USE_SONOFF_SC
#undef USE_SONOFF_SPM
#undef USE_SPI
#undef USE_TASMOTA_DISCOVERY
#undef USE_TELEGRAM
#undef USE_THERMOSTAT
#undef USE_TLS
#undef USE_TM1638
#undef USE_TUYA_MCU
#undef USE_TUYA_TIME
#undef USE_VEML6070
#undef USE_VL53L0X
#undef USE_WEBCLIENT_HTTPS
#undef USE_WEMOS_MOTOR_V1
#undef USE_WS2812
#undef USE_ZIGBEE_ZNP
#undef WEBSERVER_ADVERTISE

#endif // _USER_CONFIG_OVERRIDE_H_
```

Now i'm very confused

**Reply by s-hadinger** [reply]
Losing wifi for a period of time might also indicate a power issue with the device.

**Comment by Jason2866** [comment]
your config is wrong. You have to use as `board` the webcam so `board = esp32-cam`
Since most Cam Boards do need the PSRAM bug fix. This fix is included in the cam board definition.
The board you have choose has no PSRAM fix and Tasmota identifys at startup that PSRAM can not be used since a ESP32 is detected which needs the fix. So simply change the board and the webcam will work

**Reply by bclermont** [reply]
> your config is wrong. You have to use as board the webcam so board = esp32-cam

Well, maybe my configuration is wrong. But stock `tasmota32-webcam` do not works with the combination Freenove wrover-dev & camera.

It works terribly if I hardcode fb_count to `1`.

And it works very good with my invalid configuration, so the PSRAM worked in my misconfigured experimentation.

If stock `tasmota32-webcam` could just works would be good enough for me

**Comment by Jason2866** [comment]
Connecting anything! to the webcam can be to much for reliable work. Adding sensors is a bad idea.
Most probably your cam will work fine, if you flash the precompiled firmware and use a verified good power supply (most USB Phone chargers are NOT).
And disconnect anything!! from GPIO 0 after flashing. This is the clk of the cam. A short wire will already make trouble.

**Reply by bclermont** [reply]
> Connecting anything! to the webcam can be to much for reliable work. Adding sensors is a bad idea.

Well, the reason I pick this board was to use a combination of camera (not as a webcam, but to take a pic ever few minutes) and sensors to monitor my wood stove.

That board come with so many GPIO pins, it's kind of sad to not be able to exploit those.

**Comment by Jason2866** [comment]
Yes Gpios are free, but the ESP32 is on the edge doing picture processing. So any additional code makes
the main function (webcam) more fragil in his function.
Back to the issue, something is strange with your PSRAM. What size do you get shown in Info page for?

**Reply by bclermont** [reply]
@Jason2866 There is no PSRAM specific information

![Screenshot from 2022-03-28 14-36-16](https://user-images.githubusercontent.com/474302/160464399-82280789-afd4-4eff-a99f-a13accf8e355.png)

**Comment by Jason2866** [comment]
Okay, when you compile with `board = esp32-cam` what do you get?

**Reply by bclermont** [reply]
This is the output of `tasmota32-webcam`:

at bootup:

```
PS-RAM Memory 4093 kB
PS-RAM free Memory 4068 kB
```

If I `wcinit` it goes to:

```
PS-RAM Memory 4092 kB
PS-RAM free Memory 3324 kB
```

**Reply by Jason2866** [reply]
So  fbcount with 2 should work since there is enough PSRAM which can be used.
If not there is something wrong with the board or camera

**Comment by bclermont** [comment]
and now, camera works now with `tasmota32-webcam` without any of my changes

the only thing that I different, is that I removed the camera from the board and put it back

**Reply by Jason2866** [reply]
power supply, long wires, bad contact... Who knew 😀
